src/Workspaces/CSharp/Portable/LanguageServices/CSharpSemanticFactsService.cs

```diff
@@ -257,5 +257,20 @@ public bool IsPartial(ITypeSymbol typeSymbol, CancellationToken cancellationToke
             return SpecializedCollections.SingletonEnumerable(
                 semanticModel.GetDeclaredSymbol(memberDeclaration, cancellationToken));
         }
+
+        public SyntaxNode RemoveObjectCastIfAny(SemanticModel semanticModel, SyntaxNode node)
+        {
+            if (node is CastExpressionSyntax cast)
```

Comment:
IOperatoin could make this uniform for both languages ;-)   J/K  don't take a dependency on that now.

Reply:
However, to be nitpicky, this is not the right way to use these services.  The proper thing to do here is use (or add if necessary) an IsCastExpression helper to ISyntaxFacts service.  Then pull this method up to the abstract base class so you don't need to duplicate it for for both languages.

Here's hte general philosophy:

1. if it's code that makes VB and C# similar *language* constructs the same, then put in the 'XXXFactsService'.  
2. If it's code needed for just a single analyzer/fixer, then keep the core logic in the analyzer.
3. if it's valuable code needed for multiple analyzers/fixers, then put in a shared service somewhere.

i.e. if you needed the RemoveObjectCastIfAny for many different analyzers, then taking it out of the "use null propagation" analyzer code would be appropriate.  However, since it's just for that use case, i would keep the core logic there, and only extract out the language specific code (i.e. IsCaseExpression, and GetPartsOfCastExpression).  This matches the patterns and practices of almost all the other analyzers.

Reply:
That works. Thanks for the feedback!
A single specialized API seemed more straightforward, but I understand it offered less opportunities for re-use. Made the change to use IsCastExpression/GetPartsOfCastExpression instead.